projects/angular/src/forms/datepicker/date-input.ts

```diff
@@ -19,98 +17,62 @@ import {
   OnDestroy,
   OnInit,
   Optional,
-  Output,
   PLATFORM_ID,
   Renderer2,
   Self,
   ViewContainerRef,
 } from '@angular/core';
 import { NgControl } from '@angular/forms';
-import { of } from 'rxjs';
-import { filter, switchMap } from 'rxjs/operators';
+import { filter } from 'rxjs/operators';
 
 import { isBooleanAttributeSet } from '../../utils/component/is-boolean-attribute-set';
 import { FocusService } from '../common/providers/focus.service';
 import { WrappedFormControl } from '../common/wrapped-control';
 import { ClrDateContainer } from './date-container';
-import { DayModel } from './model/day.model';
 import { DateFormControlService } from './providers/date-form-control.service';
 import { DateIOService } from './providers/date-io.service';
 import { DateNavigationService } from './providers/date-navigation.service';
 import { DatepickerEnabledService } from './providers/datepicker-enabled.service';
 import { DatepickerFocusService } from './providers/datepicker-focus.service';
-import { datesAreEqual } from './utils/date-utils';
 
 // There are four ways the datepicker value is set
 // 1. Value set by user typing into text input as a string ex: '01/28/2015'
 // 2. Value set explicitly by Angular Forms APIs as a string ex: '01/28/2015'
 // 3. Value set by user via datepicker UI as a Date Object
 // 4. Value set via `clrDate` input as a Date Object
 
-@Directive({
-  selector: '[clrDate]',
-  host: {
-    '[class.clr-input]': 'true',
-  },
-  providers: [DatepickerFocusService],
-})
-export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implements OnInit, AfterViewInit, OnDestroy {
+@Directive()
+export class ClrDateInputBase extends WrappedFormControl<ClrDateContainer> implements OnInit, OnDestroy {
   static ngAcceptInputType_date: Date | null | string;
 
   @Input() placeholder: string;
-  @Output('clrDateChange') dateChange = new EventEmitter<Date>(false);
 
   protected override index = 1;
 
-  private initialClrDateInputValue: Date;
-  private previousDateChange: Date;
-
   constructor(
     viewContainerRef: ViewContainerRef,
     injector: Injector,
     protected override el: ElementRef<HTMLInputElement>,
     protected override renderer: Renderer2,
+    @Inject(DOCUMENT) protected document: any,
```

Comment:
This property is only needed in the start date input directive. Please move it there. You can use the `inject` function to avoid having to call the `super` constructor with all of the base class dependencies. See 207df7400f47ec1b9fa3ad705c52185da1da061a.